chain/ethereum/src/codec.rs

```diff
@@ -395,6 +395,7 @@ impl BlockHeader {
     pub fn parent_ptr(&self) -> Option<BlockPtr> {
         match self.parent_hash.len() {
             0 => None,
+            _ if self.parent_hash.iter().all(|x| *x == 0) => None,
```

Comment:
We can now remove the above `0 => None` branch, noting in a comment that this check will also return `None` if the len is 0.